src/_tutorials/language/futures.md

```diff
@@ -175,45 +174,59 @@ program prints.  If `gatherNewsReports()` takes a little while to complete its
+     alt="diagram showing flow of control through the main() and printDailyNewsDigest functions">
 
 1. The app begins executing.
-1. The main function calls `printDailyNewsDigest()`, which (because it's
-   marked `async`), immediately returns a Future, _before any code is
-   executed_.
+1. The `main()` function calls the async function `printDailyNewsDigest()`,
+   which begins executing synchronously.
+1. `printDailyNewsDigest()` uses `await` to call the async function
+   `gatherNewsReports()`, which begins executing synchronously.
+1. Because `gatherNewsReports()` is an async function, when it reaches its
+   `return` statement it pauses its execution and returns an uncompleted
+   Future (an instance of `Future<String>`).
+1. Because `printDailyNewsDigest()` is an async function and is
+   awaiting a value, it pauses its execution and returns an uncompleted
+   Future (in this case, an instance of `Future<void>`)
+   to its caller (`main()`).
 1. The remaining print functions execute. Because they're synchronous, each
    function executes fully before moving on to the next print function. For
    example, the winning lottery numbers are all printed before the weather
    forecast is printed.
-1. The body of the `printDailyNewsDigest()` function starts executing.
-
-1. After reaching the await expression (`await gatherNewsReports()`) and
-   calling `gatherNewsReports()`, the program pauses, waiting for the Future
-   returned by `gatherNewsReports()` to complete.
-1. Once that Future completes, execution of `printDailyNewsDigest()`
-   continues, printing the news.
-1. When the `printDailyNewsDigest()` function body has completed executing,
+1. When `main()` has finished executing, the asynchronous functions can resume
+   execution. First, the Future returned by `gatherNewsReports()` completes.
+   Then `printDailyNewsDigest()` continues executing, printing the news.
+1. When the `printDailyNewsDigest()` function body finishes executing,
    the Future that it originally returned completes, and the app exits.
 
-<aside class="alert alert-info" markdown="1">
-  **Note:** If an async function doesn't explicitly return a value, it returns
-  a Future wrapped around a null value.
-</aside>
+Note that an async function starts executing right away (synchronously).
+The function pauses execution and returns an uncompleted Future
+when it reaches the first occurrence of any of the following:
+
+* The function's first `await` expression (after the function
+  gets the uncompleted Future from that expression).
+* The function's `return` statement.
+* The end of the function body.
+
 
 ### Handling errors {#handling-errors-async}
 
 If a Future-returning function completes with an error, you probably want to
-capture that error.  Async functions can use try-catch to capture the error.
+capture that error. Async functions can use try-catch to capture the error.
```

Comment:
I know you only added a space here, but could this use reduced redundancy? "use try-catch to do so" ?